scripts/create_gitlab_snapshot.py

```diff
@@ -38,24 +38,51 @@
 def main(argv):
     import argparse
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument('--no-restart',
-                        dest='restart',
-                        default=True,
-                        action='store_false',
-                        help='Leave the EC2 instance in a stopped state after '
-                             'the snapshot has been created.')
+    g = parser.add_mutually_exclusive_group()
+    g.add_argument('--no-restart',
+                   dest='restart',
+                   default=True,
+                   action='store_false',
+                   help='Leave the EC2 instance in a stopped state after '
+                        'the snapshot has been created.')
+    g.add_argument('--start-only',
+                   dest='start',
+                   default=False,
+                   action='store_true',
+                   help='Start the EC2 instance from a stopped state. '
+                        'Fails if the instance is not in a stopped state.')
+    g.add_argument('--new-size',
+                   dest='size',
+                   action='store',
+                   type=int,
+                   help='After creating a new snapshot, restore the snapshot to a new'
+                        'data volume of the given size (in GiB).')
     args = parser.parse_args(argv)
     require(config.terraform_component == 'gitlab',
             "Select the 'gitlab' component ('dev.gitlab' or 'prod.gitlab', for example).")
     volume = gitlab_volume_info()
     instance: Optional[JSON] = only(volume['Attachments'])
     if instance is None:
         log.info('Volume %r is not attached to any instances', volume['VolumeId'])
+    elif args.start:
+        verify_instance_state(instance['InstanceId'])
+        start_instance(instance)
     else:
         shutdown_instance(instance)
-    create_snapshot(volume)
-    if instance and args.restart:
-        start_instance(instance)
+        snapshot_id = create_snapshot(volume)
+        if args.size:
+            create_volume(snapshot_id, args.size, volume)
+        elif args.restart:
+            start_instance(instance)
+
+
+def verify_instance_state(id: str):
+    response = aws.ec2.describe_instance_status(
+        InstanceIds=[id],
+    )
+    assert response['ResponseMetadata']['HTTPStatusCode'] == 200
+    require(not response['InstanceStatuses'],
+            f'Instance {id} is not in expected state.')
```

Comment:
What is this supposed to check for? What is the expected state?

Reply:
The list `response['InstanceStatuses']` should be `[]` when the instance is stopped (the expected state). Conversely, an active instance would have `[instance-id]` as `response['InstanceStatuses']`.
I rewrote this bit, I think the intent is a tad clearer.